feat(clerk-react,clerk-js, types,shared): Introduce `Clerk.status` (#5476)
clerk/javascript: .changeset/cool-showers-admire.md

---
'@clerk/nextjs': minor
'@clerk/clerk-react': minor
---

> Introduce `useClerk().status` alongside `<ClerkFailed/>` and `<ClerkDegraded/>`.

Reviewer: nit

```suggestion
Introduce `useClerk().status` alongside `<ClerkFailed />` and `<ClerkDegraded />`.
```

### `useClerk().status`

> Possible values are:
> - `loading` set on during initialization
> - `error` set when hotloading clerk-js failed or `Clerk.load()` failed
> - `ready` set when Clerk is fully operational
> - `degraded` set when Clerk is partially operational

Reviewer: ```suggestion
Possible values for `useClerk().status` are:
- `"loading"`: Set during initialization
- `"error"`: Set when hotloading clerk-js failed or `Clerk.load()` failed
- `"ready"`: Set when Clerk is fully operational
- `"degraded"`: Set when Clerk is partially operational
```

> 
> `useClerk().loaded` is `true` when `useClerk().status` is either `"ready"` or `"degraded"`.
> `useClerk().loaded` is `false` when `useClerk().status` is `"loading"` or `"error"`.

Reviewer: ```suggestion
The computed value of `useClerk().loaded` is:

- `true` when `useClerk().status` is either `"ready"` or `"degraded"`.
- `false` when `useClerk().status` is `"loading"` or `"error"`.
```

> ### `<ClerkFailed/>`

Reviewer: nit

```suggestion
### `<ClerkFailed />`
```

```tsx
<ClerkLoaded>
  <MyCustomSignInForm/>
</ClerkLoaded>
<ClerkFailed>
  <ContactSupportBanner/>
</ClerkFailed>
```

> ### `<ClerkDegraded/>`

Reviewer: nit

```suggestion
### `<ClerkDegraded />`
```
